src/index/index.h

```diff
@@ -28,82 +29,97 @@ typedef struct {
 	IndexType type;             // Index type exact-match / fulltext.
 } Index;
 
-// Create a new FullText index.
-Index *Index_New
-(
-	const char *label,  // Indexed label
-	IndexType type      // Index is a fulltext index
-);
-
-// Adds field to index.
-void Index_AddField
-(
-	Index *idx,
-	const char *field
-);
-
-// Removes fields from index.
-void Index_RemoveField
-(
-	Index *idx,
-	const char *field
-);
-
-// Index node.
-void Index_IndexNode
-(
-	Index *idx,     // Index to use
-	const Node *n   // Node to index
-);
-
-// Remove node from index.
-void Index_RemoveNode
-(
-	Index *idx,     // Index to use
-	const Node *n   // Node to remove
-);
-
-// Constructs index.
-void Index_Construct
-(
-	Index *idx
-);
-
-// Query index.
-RSResultsIterator *Index_Query
-(
-	const Index *idx,
-	const char *query,          // Query to execute
-	char **err                  // Optional, report back error
-);
-
-// Return indexed label.
-const char *Index_GetLabel
-(
-	const Index *idx
-);
-
-// Returns number of fields indexed.
-uint Index_FieldsCount
-(
-	const Index *idx
-);
-
-// Returns indexed fields.
-const char **Index_GetFields
-(
-	const Index *idx
-);
-
-// Checks if given field is indexed.
-bool Index_ContainsField
-(
-	const Index *idx,
-	const char *field
-);
-
-// Free fulltext index.
-void Index_Free
-(
-	Index *idx
-);
+/**
+ * @brief  Create a new index.
+ * @param  *label: Indexed label
+ * @param  type: Index type - exact match or full text.
+ * @retval New constructed index for the label.
+ */
+Index *Index_New(const char *label, IndexType type);
+
+/**
+ * @brief  Adds field to index.
+ * @param  *idx: Index
+ * @param  *field: Field to add.
+ */
+void Index_AddField(Index *idx, const char *field);
```

Comment:
Let's be consistent if `Index_RemoveField` accepts `Attribute_ID`, `Index_AddField` should do the same.